tests/Unit/Evolution/Imex/Test_SolveImplicitSector.cpp

```diff
@@ -0,0 +1,750 @@
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Literals.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace {
+// Set temporarily to verify that the solver correctly skips most of
+// the work when the step is explicit.
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+bool performing_step_with_no_implicit_term = false;
+
+struct Var1 : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+
+struct Var2 : db::SimpleTag {
+  using type = tnsr::II<DataVector, 2>;
+};
+
+struct Var3 : db::SimpleTag {
+  using type = tnsr::I<DataVector, 2>;
+};
+
+struct NonTensor : db::SimpleTag {
+  using type = double;
+};
+
+// These next several tags aren't used in the calculation, just for
+// testing DataBox handling.
+struct TensorFromEvolution : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+
+using VariablesFromEvolution = Tags::Variables<tmpl::list<TensorFromEvolution>>;
+
+struct TensorTemporary : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+
+struct VariablesTemporary : db::SimpleTag {
+  using type = Variables<tmpl::list<TensorTemporary>>;
+};
+
+struct SomeComputeTagBase : db::SimpleTag {
+  using type = double;
+};
+
+struct SomeComputeTag : SomeComputeTagBase, db::ComputeTag {
+  using base = SomeComputeTagBase;
+  using argument_tags =
+      tmpl::list<Var1, TensorFromEvolution, VariablesTemporary>;
+  static void function(
+      const gsl::not_null<double*> result, const Scalar<DataVector>& var1,
+      const Scalar<DataVector>& from_evolution,
+      const Variables<tmpl::list<TensorTemporary>>& temporary) {
+    // Check the initialization of the temporary Variables in the
+    // solver DataBox.  None of the mutators modify the object, so it
+    // should always have that state.
+    CHECK(temporary.number_of_grid_points() == 1);
+    CHECK(get(get<TensorTemporary>(temporary))[0] == 0.0);
+
+    // Check slicing
+    CHECK(get(from_evolution).size() == 1);
+    CHECK(get(from_evolution)[0] == 2.0 * get(var1)[0]);
+
+    *result = get(var1)[0] + 1.0;
+  }
+};
+
+enum class PrepId { Source, Jacobian, Shared };
+
+struct RecordPreparersForTest : db::SimpleTag {
+  using type = std::array<std::pair<Var2::type, Var3::type>, 4>;
+};
+
+template <PrepId Prep>
+struct Preparer {
+  using return_tags = tmpl::list<RecordPreparersForTest>;
+  using argument_tags = tmpl::list<Var2, Var3>;
+
+  static void apply(
+      const gsl::not_null<RecordPreparersForTest::type*> prep_run_values,
+      const Var2::type& var2, const Var3::type& var3) {
+    CHECK(not performing_step_with_no_implicit_term);
+
+    std::pair current_values{var2, var3};
+    CHECK((*prep_run_values)[static_cast<size_t>(Prep)] != current_values);
+    (*prep_run_values)[static_cast<size_t>(Prep)] = std::move(current_values);
+  }
+};
+// End stuff only used for DataBox handling
+
+// [initial_guess]
```

Comment:
Do you want this `initial_guess` to be around the `AnalyticSolution` or the `InitialGuess` class below? Right now I think it's around the `AnalyticSolution`